### DISCUSSION TITLE
How to connect to a device over WiFi?

### DISCUSSION THREAD

**Discussion by olivacc** [kurbatov/firmata4j#53] [Q&A] [answered]
Is there a library to initial IODEVICE using TCP ?. I am looking to control a Firmata device over wifi.

Thanks in advance

**Comment by kurbatov** [comment, marked as answer]
Hi @olivacc,

**firmata4j** contains a transport for that - [NetworkTransport](https://github.com/kurbatov/firmata4j/blob/master/src/main/java/org/firmata4j/transport/NetworkTransport.java).

Unfortunately, it is not present in the latest released version (v 2.3.8). But you can build the library locally from the master branch of this repository in the following way:

```sh
$ git clone git@github.com:kurbatov/firmata4j.git
$ cd firmata4j
$ mvn clean install
```

Then you can make a project of yours to be dependent from the library you just built:

```xml
        <dependency>
            <groupId>com.github.kurbatov</groupId>
            <artifactId>firmata4j</artifactId>
            <version>2.3.9-SNAPSHOT</version>
        </dependency>
```

And use the network transport when constructing the `IODevice`:

```java
IODevice device = new FirmataDevice(new NetworkTransport("192.168.1.18:4334"));
```
